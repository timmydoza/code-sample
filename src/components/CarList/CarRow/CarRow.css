.CarRow {
  border: 1px solid grey;
}

.CarRow:hover {
  background: #eee;
}