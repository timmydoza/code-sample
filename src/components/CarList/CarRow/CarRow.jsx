import React from 'react';
import styles from './CarRow.css';

const CarRow = (props) => {
  return (
    <tr className={styles.CarRow} onClick={props.clickHandler}>
      <td>{props.year}</td>
      <td>{props.make}</td>
      <td>{props.model}</td>
      <td>{props.mileage}</td>
    </tr>
  );
}

export default CarRow;