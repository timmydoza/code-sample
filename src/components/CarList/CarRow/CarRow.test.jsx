import React from 'react';
import renderer from 'react-test-renderer';
import CarRow from './CarRow';

const testCar = {
  make: 'TestMake',
  model: 'TestModel',
  year: 1999,
  mileage: 1234,
  image_url: 'test.jpg',
};

it('renders correctly', () => {
  const tree = renderer
    .create(<CarRow {...testCar} clickHandler={Function.prototype} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
