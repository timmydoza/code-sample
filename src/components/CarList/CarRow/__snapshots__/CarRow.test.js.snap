// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="CarRow"
  onClick={[Function]}
>
  <td>
    1999
  </td>
  <td>
    TestMake
  </td>
  <td>
    TestModel
  </td>
  <td>
    1234
  </td>
</tr>
`;
